src/datasets/utils/file_utils.py

```diff
@@ -190,32 +169,52 @@ def cached_path(
 
     if is_remote_url(url_or_filename):
         # URL, so get it from the cache (downloading if necessary)
+        # HF HTTP URLs are prepared as to hf:// URLs
```

Comment:
not sure how to interpret this comment

Reply:
I'll make it clearer. Basically even if someone passes a HTTP HF URL like `https://huggingface.co/datasets/.../data.csv` then it will be converted to `hf://datasets/.../data.csv`, that we can parse to get repo_id/revision/etc

Reply:
ah I see, thanks :+1: